File: frontend/src/services/deckService.ts
import type { Deck } from '@/models/Deck'; 
import {GetDeck, ListDecks, CreateDeck, DeleteDeck, UpdateDeck} from '../../wailsjs/go/service/DeckService'
import type { Card } from '@/models/Card';

export async function getDecks(): Promise<Deck[]> {
  let decks = await ListDecks()
  let result: Deck[] = []
  for (let deck of decks) {
     var cards: Card[] = []
     var d: Deck ={
      id: deck.ID,
      name: deck.Name,
     }
     result = [d,...result]
    }
  return result
  
}

export async function getDeck(id: number): Promise<Deck | null> {
  let deck = await GetDeck(id)
  if (deck != null){
    return {
      id: deck.ID,
      name: deck.Name
    }
  }
  return null
}

export async function createDeck(name: string): Promise<Deck|null> {
  let newDeck = await CreateDeck(name)
  if (newDeck){
    const deck: Deck={
      id: newDeck.ID,
      name: newDeck.Name
    }
    return deck
  }
  return null
}

export async function deleteDeck(id: number) {
  await DeleteDeck(id)
}

export async function updateDeck(id: number, name: string){
  await UpdateDeck(id, name)
}

File: internal/core/domain/service/deck_service.go
package service

import (
	"kumemori/internal/core/domain/entity"
	"kumemori/internal/core/ports"
)

type DeckService struct {
	repo ports.DeckRepository
}

func NewDeckService(repo ports.DeckRepository) *DeckService {
	return &DeckService{repo: repo}
}

func (s *DeckService) CreateDeck(name string) (*entity.Deck, error) {
	deck := &entity.Deck{
		Name: name,
	}

	if err := s.repo.CreateDeck(*deck); err != nil {
		return nil, err
	}

	return deck, nil
}

func (s *DeckService) GetDeck(id uint) (*entity.Deck, error) {
	return s.repo.ReadDeck(id)
}

func (s *DeckService) ListDecks() ([]*entity.Deck, error) {
	return s.repo.ReadDecks()
}

func (s *DeckService) DeleteDeck(id uint) error {
	return s.repo.DeleteDeck(id)
}

func (s *DeckService) UpdateDeck(id uint, name string) error {
	return s.repo.UpdateDeck(id, name)
}
